src/condor_tools/htcondor_cli/job.py

```diff
@@ -76,6 +81,41 @@ def __init__(self, logger, submit_file, **options):
                 submit_data = f.read()
             submit_description = htcondor.Submit(submit_data)
 
+            annex_name = options["annex_name"]
+            if annex_name is not None:
+                # Tag the job for use by `htcondor annex create` later.
+                submit_description["MY.TargetAnnexName"] = f'"{annex_name}"'
+
+                # Require that this job run only on the specified annex,
+                # whose name we'll verify by requiring that "the same"
+                # user requested the annex in question.
+                #
+                # We don't need to know what htcondor.Submit.queue() thinks
+                # what the Owner or Submitter of the job will be, because the
+                # AuthenticatedIdentity is determined by the tokens that we
+                # hand out.  We will hand them out in such a way that this
+                # code always gets the right answer.
+                username = getpass.getuser()
+                my_identity = f'{username}@annex.osgdev.chtc.io'
```

Comment:
We could probably play similar `{username}@{htcondor.param.get("ANNEX_USER_DOMAIN", ...)}` games here, too.

Reply:
Agreed.  I'll let this settle over the weekend and then probably just do the obvious thing.